code/game/machinery/doors/firedoor.dm

```diff
@@ -517,6 +551,11 @@
 			new /obj/item/electronics/firelock (targetloc)
 	qdel(src)
 
+/obj/machinery/door/firedoor/Moved()
+	. = ..()
+	UnregisterSignal(src, COMSIG_TURF_EXPOSE)
```

Comment:
God it's fucking gone. I wrote this up once already but apparently it's been dusted.

UnregisterSignal is a proc on /datum. It unregisters a signal from the datum it was called on (in this case implicitly, the firedoor), and takes two arguments. The first argument is the thing to stop listening for a signal from, the second is the signal to stop listening for.

I think what you want to do here is loop watched_turfs, and unregister turf exposed from each, then call `CalculateWatchedTurfs`

Reply:
alright done